ballerina/json_to_copybook_convertor.bal

```diff
@@ -244,7 +244,8 @@ class JsonToCopybookConvertor {
         // If PIC has + or -, then remove the space allocated for the sign
         string picture = dataItem.getPicture();
         expectedWholeNumberLength -= (picture.startsWith("+") || picture.startsWith("-")) is true ? 1 : 0;
-        if wholeNumber.length() > expectedWholeNumberLength {
+        string integerPart = wholeNumber.startsWith("-") ? wholeNumber.substring(1) : wholeNumber;
```

Comment:
Don't we have to handle the case with `+`?

We are doing the same check twice here. Can't we check it and do the `expectedWholeNumberLength` and the substring calculations at once? Additionally, I think we can use regex instead of `startsWith` and `substring` methods.

Reply:
- `+` sign is ignored in json.
- I didn't get this `We are doing the same check twice here. Can't we check it and do the expectedWholeNumberLength and the substring calculations at once?`
- we can use regex.

Reply:
My bad, the `startsWith` check is done twice, but they are done on two different strings. So it is fine. But still, we can use the regex instead of startsWith checks. We can use a pre-compiled pattern (if that is faster). I am +1 for this approach as well.